samples/net/sockets/echo_async/README.rst

```diff
@@ -0,0 +1,83 @@
+.. _async-sockets-echo-sample:
+
+Asynchronous Socket Echo Server
+###############################
+
+Overview
+********
+
+The sockets/echo-async sample application for Zephyr implements an
+asyncronous IPv4 TCP echo server using BSD Sockets like API with
+non-blocking sockets and ``poll()`` call. It is thus further extension
+of :ref:`sockets-echo-sample` sample. The purpose of this sample is
+still to show how it's possible to develop sockets application portable
+to both POSIX and Zephyr. As such, it is kept minimal and supports only
+IPv4 and TCP.
+
+The source code for this sample application can be found at:
+:file:`samples/net/sockets/echo_async`.
+
+Requirements
+************
+
+- :ref:`networking_with_qemu`
+- or, a board with hardware networking
+
+Building and Running
+********************
+
+Build Zephyr version of sockets/echo_async application like this:
+
+.. code-block:: console
+
+    $ cd $ZEPHYR_BASE/samples/net/sockets/echo_async
+    $ make pristine
+    $ make BOARD=<board_to_use>
+
+``board_to_use`` defaults to ``qemu_x86``. In this case, you can run the
+application in QEMU using ``make run``. If you used another BOARD, you
+will need to consult its documentation for application deployment
+instructions.
```

Comment:
add:

      You can read about Zephyr support for specific boards in the
      documentation at :ref:`boards`.